pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -587,7 +588,8 @@ def __init__(self, data, storage, payload, instclass):
         self._container = None
         self.part_schemes = OrderedDict()
         pre_select = self.data.autopart.type or DEFAULT_AUTOPART_TYPE
-        for item in AUTOPART_CHOICES:
+        autopart_choices = (c for c in AUTOPART_CHOICES if is_supported_device_type(AUTOPART_DEVICE_TYPES[c[1]]))
```

Comment:
This could be probably moved to a function.

Reply:
Agreed and moved.